experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/rawObjectNode.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { leaf, SchemaBuilder } from "../../../domains";
+import {
+	boxedIterator,
+	createRawObjectNode,
+	extractRawNodeContent,
+	rawObjectErrorMessage,
+} from "../../../feature-libraries/editable-tree-2";
+import { brand } from "../../../util";
+import { contextWithContentReadonly } from "./utils";
+
+describe("raw object nodes", () => {
+	function getRawObjectNode() {
+		const builder = new SchemaBuilder({ scope: "raw object test" });
+		const objectSchema = builder.object("object", {
+			foo: leaf.number,
+			bar: builder.optional(leaf.string),
+			baz: builder.sequence(leaf.boolean),
+		});
+		const rootFieldSchema = SchemaBuilder.required(objectSchema);
+		const schema = builder.intoSchema(rootFieldSchema);
+		const context = contextWithContentReadonly({
+			schema,
+			initialTree: { foo: 42, baz: [] },
+		});
+
+		assert(context.root.is(rootFieldSchema));
+		let node = context.root.content;
+		const rawObjectNode = createRawObjectNode(objectSchema, {
+			foo: 42,
+			bar: undefined,
+			baz: [],
+		});
+		// This assignment checks that the raw node is assignable to the same type as the real node
+		node = rawObjectNode;
+		return { rawObjectNode, objectSchema };
+	}
+
+	it("allow reading schema data", () => {
+		const { rawObjectNode, objectSchema } = getRawObjectNode();
+		assert.equal(rawObjectNode.schema, objectSchema);
+		assert.equal(rawObjectNode.type, objectSchema.name);
+	});
+
+	it("allow reading value", () => {
+		const { rawObjectNode } = getRawObjectNode();
+		assert.equal(rawObjectNode.value, undefined);
+	});
+
+	it("disallow reading most node properties", () => {
+		const { rawObjectNode, objectSchema } = getRawObjectNode();
+		assertThrowsRawNodeError(() => rawObjectNode.context);
+		assertThrowsRawNodeError(() => rawObjectNode.parentField);
+		assertThrowsRawNodeError(() => rawObjectNode.tryGetField(brand("foo")));
+		assertThrowsRawNodeError(() => rawObjectNode[boxedIterator]());
+		assertThrowsRawNodeError(() => rawObjectNode.on("changing", () => {}));
+		assertThrowsRawNodeError(() => rawObjectNode.is(objectSchema));
+		assertThrowsRawNodeError(() => rawObjectNode.treeStatus());
+		assertThrowsRawNodeError(() => rawObjectNode.localNodeKey);
+	});
+
+	it("disallow reading fields", () => {
+		const { rawObjectNode } = getRawObjectNode();
+		assertThrowsRawNodeError(() => rawObjectNode.foo);
+		assertThrowsRawNodeError(() => rawObjectNode.bar);
+		assertThrowsRawNodeError(() => rawObjectNode.baz);
+	});
+
+	it("disallow reading boxed fields", () => {
+		const { rawObjectNode } = getRawObjectNode();
+		assertThrowsRawNodeError(() => rawObjectNode.boxedFoo);
+		assertThrowsRawNodeError(() => rawObjectNode.boxedBar);
+		assertThrowsRawNodeError(() => rawObjectNode.boxedBaz);
+	});
+
+	it("expose their contents", () => {
+		const { rawObjectNode } = getRawObjectNode();
+		assert.equal(extractRawNodeContent(rawObjectNode)?.foo, 42);
+	});
+
+	it("can only have their contents read once", () => {
```

Comment:
This file had all references to "struct" renamed to "object", and the file itself was renamed. This test here is new and is the only meaningful change to this file.